### PR TITLE
Add EditText

### LINKED ISSUE #11: CheckBox, EditText etc, simple widgets

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #11.

### CODE REVIEW

File: redom/android/src/main/java/com/pushtorefresh/redom/android/recycler/Adapter.kt [hunk 1]
```diff
@@ -1,18 +1,9 @@
 package com.pushtorefresh.redom.android.recycler
 
 import android.view.ViewGroup
-import androidx.appcompat.widget.AppCompatButton
-import androidx.appcompat.widget.AppCompatCheckBox
-import androidx.appcompat.widget.AppCompatTextView
-import androidx.appcompat.widget.SwitchCompat
+import androidx.appcompat.widget.*
```
Comment (sboishtyan) [inline redom/android/src/main/java/com/pushtorefresh/redom/android/recycler/Adapter.kt]: Do we want to use stars? I think an explicit declaration is better.

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            override val clazz: Class<out View<*>>,
+            override val output: Observable<O>
+    ) : Component<O, AppCompatEditText> {
+
+        override val viewStructure = toViewStructure(this)
+
+        override fun bind(view: AppCompatEditText): Disposable {
+            val disposable = CompositeDisposable()
+
+            editTextImpl.observeClicks?.also { RxView.clicks(view).subscribe(it) }
```
Comment (sboishtyan) [inline redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: you forget to add disposables to composite disposable.

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    override fun build(): Component<O, *> {
+        return EditTextComponent(
+                this,
+                EditText::class.java,
+                Observable.merge(outputObservables)
+        )
+    }
+
+    private class EditTextComponent<O : Any>(
+            private val editTextImpl: EditTextImpl<O>,
```
Comment (sboishtyan) [inline redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: editText is mutable, it means someone can mutate output or changeText for example, do we need that behavior?

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        override val viewStructure = toViewStructure(this)
+
+        override fun bind(view: AppCompatEditText): Disposable {
+            val disposable = CompositeDisposable()
+
+            editTextImpl.observeClicks?.also { RxView.clicks(view).subscribe(it) }
+            editTextImpl.observeText?.also { RxTextView.textChanges(view).subscribe(it) }
+            editTextImpl.changeText?.also { it.subscribe(RxTextView.text(view)) }
+
+            return disposable
```
Comment (sboishtyan) [inline redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: add TODO for output.

File: redom/android/src/main/java/com/pushtorefresh/redom/android/recycler/Adapter.kt (hunk 1 above)
Comment (artem-zinnatullin) [reply redom/android/src/main/java/com/pushtorefresh/redom/android/recycler/Adapter.kt]: I agree, we can address this in separate issue now that we have codestyle #43

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt (hunk 2 above)
Comment (artem-zinnatullin) [reply redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: ouch

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt (hunk 3 above)
Comment (artem-zinnatullin) [reply redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: This raises a good question, I'll inline fields for now but that's a good point

File: redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt [hunk 5]
```diff
@@ -3,38 +3,44 @@ package com.pushtorefresh.redom.android.view
...
 
         override val viewStructure = toViewStructure(this)
 
         override fun bind(view: AppCompatEditText): Disposable {
             val disposable = CompositeDisposable()
 
-            editTextImpl.observeClicks?.also { RxView.clicks(view).subscribe(it) }
-            editTextImpl.observeText?.also { RxTextView.textChanges(view).subscribe(it) }
-            editTextImpl.changeText?.also { it.subscribe(RxTextView.text(view)) }
+            observeClicks?.also { disposable += RxView.clicks(view).subscribe(it) }
```
Comment (sboishtyan) [inline redom/android/src/main/java/com/pushtorefresh/redom/android/view/EditTextImpl.kt]: use if statement

**Pull request merged**